core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -254,16 +254,22 @@ import scala.util.control.NonFatal
   override val receive: Receive = regularReceive
 
   def regularReceive: Receive = LoggingReceive {
-    case Commit(tp, offset) =>
+    case Commit(tp, offset, emergency) =>
       // prepending, as later received offsets most likely are higher
       commitMaps = tp -> offset :: commitMaps
       commitSenders = commitSenders :+ sender()
+      if (emergency) {
+        emergencyPoll()
+      }
 
-    case CommitWithoutReply(tp, offset) =>
+    case CommitWithoutReply(tp, offset, flush) =>
       // prepending, as later received offsets most likely are higher
       commitMaps = tp -> offset :: commitMaps
+      if (flush) {
+        requestDelayedPoll()
+      }
 
-    case CommitSingle(tp, offset) =>
+    case CommitSingle(tp, offset, _) =>
```

Comment:
The `flush` parameter is not needed for `CommitSingle`.

Reply:
removed